mlrun/api/api/endpoints/functions.py

```diff
@@ -160,12 +161,44 @@ def build_status(
     tag: str = "",
     offset: int = 0,
     logs: bool = True,
+    last_time: int = 0,
+    verbose: bool = False,
     db_session: Session = Depends(deps.get_db_session),
 ):
     fn = get_db().get_function(db_session, name, project, tag)
     if not fn:
         log_and_raise(HTTPStatus.NOT_FOUND.value, name=name, project=project, tag=tag)
 
+    # nuclio deploy status
+    if fn.get("kind", "") in RuntimeKinds.serverless_runtimes():
+        state, address, nuclio_name, last_time, text = get_nuclio_deploy_status(
+            name, project, tag, last_time=last_time, verbose=verbose
+        )
+        if state == "ready":
+            logger.info(f"nuclio deploy of {name} completed successfully")
+        if state == "error":
+            logger.error(f"nuclio deploy error in {name}, {text}")
+        update_in(fn, "status.nuclio_name", nuclio_name)
+        update_in(fn, "status.state", state)
+        update_in(fn, "status.address", address)
+
+        versioned = False
+        if state == "ready":
+            versioned = True
```

Comment:
I guess you want to save the function on any terminal state, e.g. `"error"` as well.
Would be much nicer if you wouldn't use hard coded strings all over and instead `nuclio-jupyter` would expose constants for the states

Reply:
@Hedingber this part i copied from yours re the job deploy, i can make for other states, lets do the enums in future PR

Reply:
added comment